### PR TITLE
Add subgrid to key metrics widgets.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue:

- #7761

## Relevant technical choices

Implemented as in the IB, through testing I made the following additions to match the AC:
 - [Updated the number of rows to 3 per row of widgets](https://github.com/google/site-kit-wp/pull/8324/files#diff-28cb499b5e3058d0cf1ff55ad9f64eac4e7db6618558d03ca3de9c60d6cabb9fR275-R279) instead of 2 used in the IB and POC. I had to do this to align the error titles, not just the successful metrics, which looked like:

Screenshot 2024-02-27 at 13 45 15

But now look like this:

Screenshot 2024-02-27 at 14 21 52

*Note, that there is now a greater gap between the title and retry message due to the grid gap property. To remove this entirely we would need to refactor the widget errors to wrap the `h3.googlesitekit-cta__title` and `div.googlesitekit-report-error-actions` into something like a `div.googlesitekit-cta__body`. Alternatively we could remove the grid gap entirely and add margin to the elements within the grid.*

 - [Removed the `overflow: hidden` and added `display: flex` to `.googlesitekit-report-error-actions`](https://github.com/google/site-kit-wp/pull/8324/files#diff-28cb499b5e3058d0cf1ff55ad9f64eac4e7db6618558d03ca3de9c60d6cabb9fR230-L230) to correctly show the retry test/CTA.
 - [Added a row and column span to the `.googlesitekit-cta`](https://github.com/google/site-kit-wp/pull/8324/commits/8cbb1e4695e3faf89eac5f4ebc942584b0c88868#diff-f157bf19f656b132dc081eb5063ffabd0103e4a328526892e9b747fb00f70c23) so that it takes up the entire widget area.
 - [Fixed the "Maybe Later" CTA expanding to the full width of the column.](https://github.com/google/site-kit-wp/pull/8324/commits/8dfa61fda8ddc5825b9a435c1eabfe8961ed799f)

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/sass/components/key-metrics/_googlesitekit-km-widget-tile.scss [hunk 1]
```diff
@@ -263,4 +265,118 @@
...
+		}
+
+		.googlesitekit-cta--error,
+		.googlesitekit-km-widget-tile--error {
+			@media (min-width: $bp-tablet) {
+				display: grid;
+				grid-column: 1 / -1;
+				grid-row: 1 / -1;
+
+				// grid-template-columns: subgrid;
```
Comment (benbowler) [inline assets/sass/components/key-metrics/_googlesitekit-km-widget-tile.scss:314]:
```suggestion
```

**Comment by benbowler** [comment]
When reviewing, check my detailed notes in the [**Relevant technical choices** section above](https://github.com/google/site-kit-wp/pull/8324#issue-2156405131).

**Comment by benbowler** [comment]
VRTs pass on local but failing here with `Error: Failed to launch the browser process!`.

**Review by tofumatt — APPROVED** [review]
I think this is a great solution, thanks for catching the issues with IB and improving upon them.

Also tested it a bunch in different sizes and all seems good 👍🏻

**Pull request merged**